orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -182,6 +188,9 @@ void addObjectMethodsForTraining(py::module& m) {
       .def_readwrite("partition_optimizer", &TrainingParameters::partition_optimizer)
       .def_readwrite("enable_grad_norm_clip", &TrainingParameters::enable_grad_norm_clip);
 
+  py::class_<TrainingConfigurationResult> config_result(m, "TrainingConfigurationResult", "pbdoc(Configuration result for training.)pbdoc");
+  config_result.def(py::init())
+      .def_readonly("loss_scale_input_name", &TrainingConfigurationResult::loss_scale_input_name);
```

Comment:
@liqun is it possible to have this return None if mixed precision is not enabled? wasn't sure how to do that

Reply:
my understanding is that this code sets up the binding between C++ and Python constructs - so at this point we don't know whether mixed precision is enabled or not. let me know if i'm missing something

Reply:
You are right, it's been sometime since I worked with pybind, but you can use [lambdas](https://pybind11.readthedocs.io/en/stable/classes.html#binding-lambda-functions)


`.def_readonly('loss_scale_input_name', [](const TrainingConfigurationResult &res) {     
    if (res.mixed_precision_config_result.has_value()) {
        return res.loss_scale_input_name;
    } else {
        return None;
}`

That is a similar test from orttraining/orttraining/python/orttraining_pybind_state.cc, line 161

However, if we could use nullptr on C++ instead of "" or std::string(){}, this translation would be automatic in the current binding

Reply:
Agreed. Probably None is better than an empty string?

Reply:
ok, figured out how to do that. thanks for the input